lib/puppet/type/cisco_route_map.rb

```diff
@@ -0,0 +1,1642 @@
+
+    newvalues(:true, :false, :default)
+  end # property match_community_exact_match
+
+  newproperty(:match_evpn_route_type_1) do
+    desc 'Enable match BGP EVPN route type-1'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_1
+
+  newproperty(:match_evpn_route_type_2_all) do
+    desc 'Enable match all BGP EVPN route in type-2'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_2_all
+
+  newproperty(:match_evpn_route_type_2_mac_ip) do
+    desc 'Enable match mac-ip BGP EVPN route in type-2'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_2_mac_ip
+
+  newproperty(:match_evpn_route_type_2_mac_only) do
+    desc 'Enable match mac-only BGP EVPN route in type-2'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_2_mac_only
+
+  newproperty(:match_evpn_route_type_3) do
+    desc 'Enable match BGP EVPN route type-3'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_3
+
+  newproperty(:match_evpn_route_type_4) do
+    desc 'Enable match BGP EVPN route type-4'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_4
+
+  newproperty(:match_evpn_route_type_5) do
+    desc 'Enable match BGP EVPN route type-5'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_5
+
+  newproperty(:match_evpn_route_type_6) do
+    desc 'Enable match BGP EVPN route type-6'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_6
+
+  newproperty(:match_evpn_route_type_all) do
+    desc 'Enable match BGP EVPN route type 1-6'
+
+    newvalues(:true, :false, :default)
+  end # property match_evpn_route_type_all
+
+  newproperty(:match_ext_community, array_matching: :all) do
+    format = '[ecomm1, ecomm2]'
+    desc 'Match BGP extended community list. An array of [ecomm1, ecomm2 and so on]' \
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
+      value.inspect
+    end
+
+    def is_to_s(value)
+      value.inspect
+    end
+
+    munge do |value|
+      begin
+        return value = :default if value == 'default'
+        value
+      end
+    end
+  end # property match_ext_community
+
+  newproperty(:match_ext_community_exact_match) do
+    desc 'Enable exact matching of extended communities'
+
+    newvalues(:true, :false, :default)
+  end # property match_ext_community_exact_match
+
+  newproperty(:match_interface, array_matching: :all) do
+    format = '[int1, int2]'
+    desc 'Match first hop interface of route. An array of [int1, int2 and so on]' \
```

Comment:
Does order matter for this property?  Are `['loopback2', 'mgmt0']` and `['mgmt0', 'loopback2']` identical in terms of the behavior of this property?